### ISSUE TITLE
GRPO.py  per_token_loss error?

### ISSUE THREAD

**Issue by ythwilam** [aburkov/theLMbook#3]
At line 263, when calculating per_token_loss, do you think we should use ref_token_log_probs.detach() instead of token_log_probs.detach()?

**Comment by aburkov** [comment]
It's a "hack" that allows us not to store the copy of the policy from the previous step. Storing it would consume more memory so we couldn't train even a 0.5B model.
